zmon_agent/discovery/kubernetes/cluster.py

```diff
@@ -505,28 +508,34 @@ def list_postgres_databases(*args, **kwargs):
         return []
 
 
-def get_postgresql_clusters(kube_client, cluster_id, alias, environment, region, infrastructure_account,
+def get_postgresql_clusters(statefulsets, kube_client, cluster_id, alias, environment, region, infrastructure_account,
                             hosted_zone, namespace=None):
 
+    ssets = [ss for ss in statefulsets]
+
     # TODO in theory clusters should be discovered using CRDs
     services = get_all(kube_client, kube_client.get_services, namespace)
 
     for service in services:
         obj = service.obj
 
-        # TODO: filter in the API call
         labels = obj['metadata'].get('labels', {})
-        if labels.get('application') != 'spilo':
+        version = labels.get('version')
+
+        # we skip non-Spilos and replica services
+        if labels.get('application') != 'spilo' or labels.get('spilo-role') == 'replica':
             continue
 
         service_namespace = obj['metadata']['namespace']
         service_dns_name = '{}.{}.svc.cluster.local'.format(service.name, service_namespace)
 
-        # for master we do not use selector at all
-        is_replica = True if obj["spec"].get("selector") else False
+        statefulset = [ss for ss in ssets if ss['version'] == version]
+        if not statefulset:  # can happen when the replica count is 0.In this case we don't have a running cluster.
+            continue
```

Comment:
please do create the entity and add a field indicating that there is no statefulset. This will be similar to the CRD discovery. That should also indicate that there is no service or no statefulset to monitor operation of the operator.